recipes/cairo/meson/conanfile.py

```diff
@@ -138,100 +142,93 @@ def validate(self):
         else:
-            defs["xcb"] = "disabled"
-            defs["xlib"] = "disabled"
+            options["xcb"] = "disabled"
+            options["xlib"] = "disabled"
         if self.options.get_safe("with_opengl") == "desktop":
-            defs["gl-backend"] = "gl"
+            options["gl-backend"] = "gl"
         elif self.options.get_safe("with_opengl") == "gles2":
-            defs["gl-backend"] = "glesv2"
+            options["gl-backend"] = "glesv2"
         elif self.options.get_safe("with_opengl") == "gles3":
-            defs["gl-backend"] = "glesv3"
+            options["gl-backend"] = "glesv3"
         else:
-            defs["gl-backend"] = "disabled"
-        defs["glesv2"] = boolean(self.options.get_safe("with_opengl") == "gles2")
-        defs["glesv3"] = boolean(self.options.get_safe("with_opengl") == "gles3")
-        defs["tee"] = boolean(self.options.tee)
-        defs["symbol-lookup"] = boolean(self.options.get_safe("with_symbol_lookup"))
+            options["gl-backend"] = "disabled"
+        options["glesv2"] = is_enabled(self.options.get_safe("with_opengl") == "gles2")
+        options["glesv3"] = is_enabled(self.options.get_safe("with_opengl") == "gles3")
+        options["tee"] = is_enabled(self.options.tee)
+        options["symbol-lookup"] = is_enabled(self.options.get_safe("with_symbol_lookup"))
 
         # future options to add, see meson_options.txt.
         # for now, disabling explicitly, to avoid non-reproducible auto-detection of system libs
-        defs["cogl"] = "disabled"  # https://gitlab.gnome.org/GNOME/cogl
-        defs["directfb"] = "disabled"
-        defs["drm"] = "disabled" # not yet compilable in cairo 1.17.4
-        defs["openvg"] = "disabled"  # https://www.khronos.org/openvg/
-        defs["qt"] = "disabled" # not yet compilable in cairo 1.17.4
-        defs["gtk2-utils"] = "disabled"
-        defs["spectre"] = "disabled"  # https://www.freedesktop.org/wiki/Software/libspectre/
-
-        meson.configure(
-            source_folder=self._source_subfolder,
-            args=["--wrap-mode=nofallback"],
-            build_folder=self._build_subfolder,
-            defs=defs,
-        )
-        return meson
+        options["cogl"] = "disabled"  # https://gitlab.gnome.org/GNOME/cogl
+        options["directfb"] = "disabled"
+        options["drm"] = "disabled"  # not yet compilable in cairo 1.17.4
+        options["openvg"] = "disabled"  # https://www.khronos.org/openvg/
+        options["qt"] = "disabled"  # not yet compilable in cairo 1.17.4
+        options["gtk2-utils"] = "disabled"
+        options["spectre"] = "disabled"  # https://www.freedesktop.org/wiki/Software/libspectre/
+
+        meson = MesonToolchain(self)
+        meson.project_options.update(options)
+
+        if is_apple_os(self) and Version(self.version) <= "1.17.4":
+            # This was fixed in the meson build from 1.17.6
```

Comment:
does not exactly match comment?

```suggestion
        if is_apple_os(self) and Version(self.version) < "1.17.6":
            # This was fixed in the meson build from 1.17.6
```